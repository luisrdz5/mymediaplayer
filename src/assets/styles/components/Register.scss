.registro {
    background: linear-gradient(#21C08B, #5630fb);
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    padding: 0px 30px;
    min-height: calc(100vh - 200px);
  }
  
  .registro__container {
    background-color: rgba(255, 255, 255, 0.1);
    border: 2px solid white;
    border-radius: 40px;
    color: white;
    padding: 60px 68px 40px;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
  }
  
  .registro__container--form {
    display: flex;
    flex-direction: column;
  }
  
  .registro__container--form label {
    font-size: 14px;
  }
  
  .registro__container--register {
    font-size: 14px;
    text-align: center;
  }
  
  .registro__container--register a {
    font-size: 16px;
    color: white;
    font-weight: bold;
    text-decoration: none;
  }
  
  .registro__container--register a:hover {
    text-decoration: underline;
  }
  