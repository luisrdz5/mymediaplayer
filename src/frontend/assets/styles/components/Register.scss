.registro {
    background: linear-gradient(#21C08B, #5630fb);
    display: flex;
    align-items: center;
    flex-direction: column;
    justify-content: center;
    padding: 0px 30px;
    min-height: calc(100vh - 200px);
  }
  
  .registro__container {
    background-color: rgba(255, 255, 255, 0.1);
    width: 450px;
    border: 2px solid white;
    border-radius: 40px;
    box-shadow: 0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22);
    color: white;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    padding: 60px 68px 40px;
    text-align: center;
  }

  .registro__container button {
    width: 100%;
  }
  .registro__container input {
    margin: 8px;
    width: 100%;
  }
  
  .registro__container--form {
    display: flex;
    flex-direction: column;
  }
  
  .registro__container--form label {
    font-size: 14px;
  }
  
  .registro__container--register {
    font-size: 14px;
    text-align: center;
  }
  
  .registro__container--register a {
    font-size: 16px;
    color: white;
    font-weight: bold;
    text-align: center;
    text-decoration: none;
  }

  .registro__container--form {
    margin: 8px;
  }
  
  .registro__container--register a:hover {
    text-decoration: underline;
  }
  