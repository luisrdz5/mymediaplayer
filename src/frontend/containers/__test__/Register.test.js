import React from 'react';
import { create } from 'react-test-renderer';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ProviderMock from '../../__mocks__/providerMock';
import Register from '../Registro';

configure({ adapter: new Adapter() });

describe('Register component', () => {

  test('Match Snapshot', () => {
    const register = create(
      <ProviderMock>
        <Register />
      </ProviderMock>,
    );
    expect(register.toJSON()).toMatchSnapshot();
  });
  it('Calls and executes preventDefault function onSubmit form', () => {
    const preventDefault = jest.fn();
    const wrapper = mount(
      <ProviderMock>
        <Register />
      </ProviderMock>,
    );
    wrapper.find('form').simulate('submit', { preventDefault });
    expect(preventDefault).toHaveBeenCalledTimes(1);
    wrapper.unmount();
  });
});
