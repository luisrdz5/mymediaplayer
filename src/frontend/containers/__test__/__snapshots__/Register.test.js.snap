// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component Match Snapshot 1`] = `
Array [
  <section
    className="registro"
  >
    <section
      className="registro__container"
    >
      <h2>
        Regístrate
      </h2>
      <form
        action="registro__container--form"
        onSubmit={[Function]}
      >
        <input
          className="input"
          name="name"
          onChange={[Function]}
          placeholder="Nombre"
          required={true}
          type="text"
        />
        <input
          className="input"
          name="email"
          onChange={[Function]}
          placeholder="Correo"
          required={true}
          type="text"
        />
        <input
          className="input"
          name="password"
          onChange={[Function]}
          placeholder="Contraseña"
          required={true}
          type="password"
        />
        <button
          className="button"
          type="submit"
        >
          Registrarme
        </button>
      </form>
      <p
        className="registro__container--register"
      >
        <a
          href="/login"
          onClick={[Function]}
        >
          Iniciar Sesión
        </a>
      </p>
    </section>
  </section>,
  <footer
    className="footer blue"
  >
    <a
      href="./"
    >
      Terminos de uso
    </a>
    <a
      href="./"
    >
      Declaración de Privacidad
    </a>
    <a
      href="./"
    >
      Centro de Ayuda
    </a>
  </footer>,
]
`;
